editor/hooks/use-pending-callback.tsx

```
@@ -0,0 +1,72 @@
+ * @example
+ * ```tsx
+ * const [state, handlers] = usePendingCallback();
+ *
+ * // Set a callback to be executed later
+ * handlers.setCallback(() => () => console.log('Hello'));
+ *
+ * // Execute the callback when ready
+ * await handlers.executeCallback();
+ * ```
+ */
+function usePendingCallback<T extends Function = Function>(): [
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)